src/Domain/Common/Validators/CustomValidators.cs

```diff
@@ -0,0 +1,26 @@
+﻿using FluentValidation;
+
+namespace CleanCompanyName.DDDMicroservice.Domain.Common.Validators;
+
+internal static class CustomValidators
+{
+    internal static IRuleBuilderOptions<T, TProperty> WithCustomErrorCode<T, TProperty>(this IRuleBuilderOptions<T, TProperty> ruleBuilder, AbstractValidator<T> validator)
+    {
+        var type = typeof(T);
+        var memberName = type.Name;
+
+        var validationRule = validator.LastOrDefault();
+
+        var propertyName = validationRule?.PropertyName;
+
+        var ruleComponent = validationRule?.Components.LastOrDefault();
+        var validationType = ruleComponent?.Validator.Name.Replace("Validator", "");
+
+        return ruleBuilder.WithErrorCode($"{memberName}_{propertyName}_{validationType}");
```

Comment:
The `null` from `.LastOrDefault()` is handled such that the error code will be `$"{memberName}__"`. Is this really intended?

Reply:
I will add a check to skip the whole custom error code if we get the validationRule null.